src/services/recovery/recovery-state.ts

```diff
@@ -7,16 +7,11 @@ import type { JsonRpcProvider } from '@ethersproject/providers'
 import type { TransactionReceipt } from '@ethersproject/abstract-provider'
 
 import type { RecoveryQueueItem, RecoveryState } from '@/store/recoverySlice'
+import { hexZeroPad } from 'ethers/lib/utils'
 
 const MAX_PAGE_SIZE = 100
 
-export const _getQueuedTransactionsAdded = (
-  transactionsAdded: Array<TransactionAddedEvent>,
-  txNonce: BigNumber,
-): Array<TransactionAddedEvent> => {
-  // Only queued transactions with queueNonce >= current txNonce
-  return transactionsAdded.filter(({ args }) => args.queueNonce.gte(txNonce))
-}
+export const normalizeTxServiceUrl = (url: string): string => (url.endsWith('/') ? url : `${url}/`)
```

Comment:
Nit: there is a `trimTrailingSlash` function in `@/utils/url.ts` that we could export and use instead of this. We would then have to append the `/` within `_getSafeCreationReceipt`.